code/datums/components/storage.dm

```diff
@@ -0,0 +1,335 @@
+// STORAGE COMPONENT
+// example:
+// /obj/item/storage/backpack/monkey
+//	 spawn_contents = list(/obj/item/toy/plush/small/monkey)
+//	 New()
+//		 ..()
+//		 AddComponent(/datum/component/storage, max_wclass = 4)
+// since /obj/item/storage already calls AddComponent with spawn_contents = spawn_contents, we don't need to specify that arg again. we just need to respecify our unique max_wclass of 4, or else it'll be set as /obj/item/storage/backpack's 3!
+
+/datum/component/storage
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS // old component gets passed new component's initialization args
+	var/datum/hud/storage/hud
+	// future: add stored_contents list and use that to keep track instead of source.contents
+	var/list/can_hold // if this exists, the storage checks it for what can be held
+	var/in_list_or_max = FALSE // if this is true, the storage allows items that are either in its can_hold list or items that have a lesser w_class than max_wclass
+	var/max_wclass = 2
+	var/slots = 7
+	var/sneaky = FALSE // does it play a sound when used?
+	var/does_not_open_in_pocket = TRUE
```

Comment:
Negatives are notoriously confusing in variable names / meanings. Maybe it's worth to refactor this to `opens_in_pocket`?

Reply:
Done.